decimalSuite_test.go

```diff
@@ -93,32 +137,45 @@ func isRoundingErr(res, expected Decimal64) bool {
 }
 
 // getInput gets the test file and extracts test using regex, then returns a map object and a list of test names.
-func getInput(file string) (data []testCaseStrings) {
-	r := regexp.MustCompile(`(?:\n)` + // start with newline (?: non capturing group)
+func getInput(line string) testCaseStrings {
+	testRegex := regexp.MustCompile(
 		`(?P<testName>dd[\w]*)` + // first capturing group: testfunc made of anything that isn't a whitespace
-		`(?:\s*)` + // match any whitespace (?: non capturing group)
-		`(?P<testFunc>[\S]*)` + // testfunc made of anything that isn't a whitespace
-		`(?:\s*\'?)` + // after can be any number of spaces and quotations if they exist (?: non capturing group)
-		`(?P<val1>\+?-?[^\t\f\v\' ]*)` + // first test val is anything that isnt a whitespace or a quoteation mark
-		`(?:'?\s*'?)` + // match any quotation marks and any space (?: non capturing group)
-		`(?P<val2>\+?-?[^\t\f\v\' ]*)` + // second test val is anything that isnt a whitespace or a quoteation mark
-		`(?:'?\s*'?)` +
-		`(?P<val3>\+?-?[^->]?[^\t\f\v\' ]*)` + //testvals3 same as 1 but specifically dont match with '->'
-		`(?:'?\s*->\s*'?)` + // matches the indicator to answer and surrounding whitespaces (?: non capturing group)
-		`(?P<expectedResult>\+?-?[^\r\n\t\f\v\' ]*)`) // matches the answer that's anything that is plus minus but not quotations
+			`(?:\s*)` + // match any whitespace (?: non capturing group)
```

Comment:
nit: No need to explain the non-capturing bit. If anyone wants to understand the meaning of a regexp construct, they should read the documentation.